openmetadata-service/src/main/java/org/openmetadata/service/apps/AppRepository.java

```diff
@@ -0,0 +1,167 @@
+  public static String APP_SCHEDULE_EXTENSION = "ScheduleExtension";
+  public static String APP_SCHEDULE_INFO_SCHEMA = "scheduleInfo.json";
+
+  public AppRepository(CollectionDAO dao) {
+    super(AppResource.COLLECTION_PATH, Entity.APPLICATION, Application.class, dao.applicationDAO(), dao, "", "");
+    supportsSearchIndex = false;
+  }
+
+  @Override
+  public Application setFields(Application entity, EntityUtil.Fields fields) {
+    entity.setSchedules(fields.contains("schedules") ? getApplicationSchedule(entity) : null);
+    entity.setPipelines(fields.contains("pipelines") ? getIngestionPipelines(entity) : entity.getPipelines());
+    return entity.withBot(getBotUser(entity));
+  }
+
+  private List<AppSchedule> getApplicationSchedule(Application app) {
+    List<CollectionDAO.ExtensionRecord> schedules =
+        daoCollection.entityExtensionDAO().getExtensions(app.getId(), APP_SCHEDULE_EXTENSION);
+    return schedules.stream()
+        .map((schedule) -> JsonUtils.readValue(schedule.getExtensionJson(), AppSchedule.class))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public Application clearFields(Application entity, EntityUtil.Fields fields) {
+    return entity;
+  }
+
+  @Override
+  public void prepare(Application entity, boolean update) {}
+
+  @Override
+  public void storeEntity(Application entity, boolean update) {
+    EntityReference botUserRef = entity.getBot();
+    EntityReference ownerRef = entity.getOwner();
+    entity.withBot(null).withOwner(null);
+
+    // Store
+    store(entity, update);
+
+    // Restore entity fields
+    entity.withBot(botUserRef).withOwner(ownerRef);
+  }
+
+  public EntityReference getBotUser(Application application) {
+    return application.getBot() != null
+        ? application.getBot()
+        : getToEntityRef(application.getId(), Relationship.HAS, Entity.BOT, false);
+  }
+
+  public RestUtil.PutResponse<?> addApplicationSchedule(UriInfo uriInfo, UUID appId, AppSchedule appScheduleInfo) {
+    // Get Application
+    Application application = get(uriInfo, appId, getFields("schedules"));
+    if (application.getSchedules() != null && !application.getSchedules().isEmpty()) {
+      throw new IllegalArgumentException("Job has already been Scheduled please remove any existing schedules.");
+    }
+
+    daoCollection
+        .entityExtensionDAO()
+        .insert(
+            appId,
+            String.format("%s.%s", APP_SCHEDULE_EXTENSION, appScheduleInfo.getScheduleId().toString()),
+            APP_SCHEDULE_INFO_SCHEMA,
+            JsonUtils.pojoToJson(appScheduleInfo));
+    application.getSchedules().add(appScheduleInfo);
+
+    ChangeDescription change = addScheduleChangeDescription(application.getVersion(), appScheduleInfo);
+    ChangeEvent changeEvent =
+        getChangeEvent(withHref(uriInfo, application), change, entityType, application.getVersion());
+
+    // Schedule the application to scheduler
+    ApplicationHandler.scheduleApplication(
```

Comment:
Application should have IngestionPipeline has a relation. If the application is  external this should deploy the pipeline